### PR TITLE
feat(annotation): 接入dataMaker dataRegion regionFilter组件

### PR DESCRIPTION
**Pull request by zqlu**
### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [ ] All tests pass and/or benchmarks are included
- [ ] Commit message follows commit guidelines
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by zqlu** [comment]
依赖组件部分的MR： https://github.com/antvis/component/pull/121

File: src/chart/controller/annotation.ts [hunk 1]
```diff
@@ -77,6 +79,51 @@ export interface TextOption {
...
+  readonly content: string | number;
+  /** 文本的图形样式属性 */
+  readonly style?: object;
+  /** 是否进行动画 */
+  readonly animate?: boolean;
+  /** x 方向的偏移量 */
+  readonly offsetX?: number;
+  /** y 方向的偏移量 */
+  readonly offsetY?: number;
+}
```
Comment (hustcc) [inline src/chart/controller/annotation.ts]: 可以继承 BaseOption？

File: src/chart/controller/annotation.ts [hunk 2]
```diff
@@ -161,23 +231,38 @@ export default class Annotation extends Controller<BaseOption[]> {
...
-        newCache.set(key, value);
-      } else {
-        // 不存在，则是所有需要被销毁的组件
-        value.component.destroy();
-      }
+      // 处理完成之后，销毁删除的
+      // 不在处理中的
+      const newCache = new Map<BaseOption, ComponentOption>();
+
+      this.cache.forEach((value: ComponentOption, key: BaseOption) => {
```
Comment (hustcc) [inline src/chart/controller/annotation.ts]: 遍历 this.getComponents() 吧~

File: src/chart/controller/annotation.ts (hunk 1 above)
Comment (zqlu) [reply src/chart/controller/annotation.ts]: 新增两个定位类型的BaseOptions

**Pull request merged**